File: VB/TreeListFilter/XtraForm1.vb
Imports System
Imports DevExpress.XtraEditors

Namespace TreeListFilter
	Partial Public Class XtraForm1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub XtraForm1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Me.DepartmentsTableAdapter.Fill(Me.DepartmentsDataSet.Departments)
            filterTreeList1.ExpandAll()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			filterTreeList1.ColumnFilterConditions.SaveToXml("..\..\filters.xml")
		End Sub

		Private Sub simpleButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton2.Click
			filterTreeList1.ColumnFilterConditions.RestoreFromXml("..\..\filters.xml")
		End Sub
	End Class
End Namespace

File: CS/TreeListFilter/XtraForm1.cs
using System;
using DevExpress.XtraEditors;

namespace TreeListFilter
{
	public partial class XtraForm1 : XtraForm
	{
		public XtraForm1()
		{
			InitializeComponent();
		}

		private void XtraForm1_Load(object sender, EventArgs e)
		{
			this.departmentsTableAdapter.Fill(this.departmentsDataSet.Departments);
			filterTreeList1.ExpandAll();
		}

		private void simpleButton1_Click(object sender, EventArgs e)
		{
			filterTreeList1.ColumnFilterConditions.SaveToXml("..\\..\\filters.xml");
		}

		private void simpleButton2_Click(object sender, EventArgs e)
		{
			filterTreeList1.ColumnFilterConditions.RestoreFromXml("..\\..\\filters.xml");
		}
	}
}